### PR TITLE
Add agreements section to package show details page

### PR DESCRIPTION
**Pull request by Yurii-Danylenko**
https://issues.folio.org/browse/UIEH-628

## Purpose
Add new feature for attaching agreements to the package. Apply a new approach of managing data in eholdings app.

## Approach
- Add an accordion for with a list of agreements
- Add actions, reducers, epics for fetching and updating agreements
- Add a "ui-find-agreements" plugin for attaching an agreement to the package
- Add tests

#### TODOS
Full agreements functionality should be tested after an ERM with agreements has been added to the FOLIO snapshot build.
Unit tests for new redux stuff will be added during work on https://issues.folio.org/browse/UIEH-630

## Screenshots
Снимок экрана 2019-03-11 в 12 10 18
Снимок экрана 2019-03-11 в 12 09 57

### CODE REVIEW

File: src/components/agreements-list/agreements-list.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        isEmptyMessage={<FormattedMessage id="ui-eholdings.agreements.notFound" />}
+        rowFormatter={this.rowFormatter}
+      />
+    );
+  }
+
+  render() {
+    return (
+      <FormattedMessage id="ui-eholdings.agreements">
+        {(ariaLabel) => this.renderMultiColumnList(ariaLabel)}
```
Comment (MaksymDryha) [inline src/components/agreements-list/agreements-list.js]: I don't think that `renderMultiColumnList` method helps to abstract from unnecessary details. `AgreementsList` is nothing but a `MultiColumnList`, so `MultiColumnList` is not a detail which should be hidden in a separate method. That's why the `renderMultiColumnList` method seems superfluous to me.

Comment (Godlevskyi) [reply src/components/agreements-list/agreements-list.js]: Agreed, no sense in a separate method

File: src/components/package/show/package-show.js [hunk 2]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+                        <Icon icon="spinner-ellipsis" />
+                      </div>
+                    )
+                }
+              </div>
+            </KeyValue>
+          )
+        }
+        {
+          hasProxy && proxyTypes.request.isResolved && model.isLoaded && provider.isLoaded
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: please move it as a constant

File: src/components/package/show/package-show.js [hunk 3]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+                    model={model}
+                    tags={getTagLabelsArr(tagsModel)}
+                  />
+                )
+              }
+            </Accordion>
+          )
+        }
+        <Accordion
+          label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.label.holdingStatus" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: Please format it

File: src/components/package/show/package-show.js [hunk 4]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+          onToggle={this.handleSectionToggle}
+        >
+          <SelectionStatus
+            model={model}
+            onAddToHoldings={this.props.addPackageToHoldings}
+          />
+        </Accordion>
+
+        <Accordion
+          label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.label.packageInformation" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: please format it

File: src/components/package/show/package-show.js [hunk 5]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+          open={sections.packageShowInformation}
+          id="packageShowInformation"
+          onToggle={this.handleSectionToggle}
+        >
+          <KeyValueColumns>
+            <div>
+              <KeyValue label={<FormattedMessage id="ui-eholdings.package.provider" />}>
+                <div data-test-eholdings-package-details-provider>
+                  <InternalLink to={`/eholdings/providers/${model.providerId}`}>{model.providerName}</InternalLink>
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: please format it

File: src/components/package/show/package-show.js [hunk 6]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+                <div data-test-eholdings-package-details-titles-total>
+                  <FormattedNumber value={model.titleCount} />
+                </div>
+              </KeyValue>
+            </div>
+          </KeyValueColumns>
+        </Accordion>
+
+        <Accordion
+          label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.package.packageSettings" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: please format it

File: src/components/package/show/package-show.js [hunk 7]
```diff
@@ -135,76 +147,438 @@ class PackageShow extends Component {
...
+        >
+          {
+            packageSelected
+              ? this.renderPackageSettings()
+              : <p><FormattedMessage id="ui-eholdings.package.visibility.notSelected" /></p>
+          }
+        </Accordion>
+
+        <Accordion
+          label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.package.coverageSettings" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/package/show/package-show.js]: please format it

File: src/components/provider/show/provider-show.js [hunk 8]
```diff
@@ -108,29 +108,189 @@ class ProviderShow extends Component {
...
     } = this.props;
-    let { sections } = this.state;
-    let hasProxy = model.proxy && model.proxy.id;
-    let hasToken = model.providerToken && model.providerToken.prompt;
+
+    const {
+      sections,
+    } = this.state;
+
+    const hasProxy = model.proxy && model.proxy.id;
```
Comment (Godlevskyi) [inline src/components/provider/show/provider-show.js]: To be consistent refactor it using hasIn operator

File: src/components/provider/show/provider-show.js [hunk 9]
```diff
@@ -108,29 +108,189 @@ class ProviderShow extends Component {
...
     const hasProviderSettings = hasProxy || hasToken;
     const hasSelectedPackages = model.packagesSelected > 0;
 
     return (
       <div>
-        <Toaster toasts={this.toasts} position="bottom" />
+        {
+          hasSelectedPackages && (
+            <Accordion
+              label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.tags" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/provider/show/provider-show.js]: Please format it

File: src/components/provider/show/provider-show.js [hunk 10]
```diff
@@ -108,29 +108,189 @@ class ProviderShow extends Component {
...
+                    model={model}
+                    tags={getTagLabelsArr(tagsModel)}
+                  />
+                )
+              }
+            </Accordion>
+          )
+        }
+        <Accordion
+          label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.provider.providerInformation" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/provider/show/provider-show.js]: Please format it

File: src/components/provider/show/provider-show.js [hunk 11]
```diff
@@ -108,29 +108,189 @@ class ProviderShow extends Component {
...
+
+          <KeyValue label={<FormattedMessage id="ui-eholdings.provider.totalPackages" />}>
+            <div data-test-eholdings-provider-details-packages-total>
+              <FormattedNumber value={model.packagesTotal} />
+            </div>
+          </KeyValue>
+        </Accordion>
+        {hasProviderSettings && (
+          <Accordion
+            label={<Headline size="large" tag="h3"><FormattedMessage id="ui-eholdings.provider.providerSettings" /></Headline>}
```
Comment (Godlevskyi) [inline src/components/provider/show/provider-show.js]: Please format it

File: src/components/provider/show/provider-show.js [hunk 12]
```diff
@@ -108,29 +108,189 @@ class ProviderShow extends Component {
...
+            open={sections.providerShowProviderSettings}
+            id="providerShowProviderSettings"
+            onToggle={this.handleSectionToggle}
+          >
+            {
+              hasProxy && (
+                !proxyTypes.request.isResolved || !rootProxy.request.isResolved || model.isLoading
```
Comment (Godlevskyi) [inline src/components/provider/show/provider-show.js]: Please move it to the constants

**Pull request merged**